src/terminal.ts

```diff
@@ -19,7 +20,17 @@ class Terminal {
     // Set holding all terminals available at any moment.
     this.terminalsSet = new Set()
 
+    // set start command asyncronously
+    // TODO does any part of the code rely on this? If so we should await the promise before there
+    setShellStartCommand().catch(() => {
+      atom.config.set("terminal.shell", getFallbackShell())
+    })
+
     this.disposables.add(
+      // an observre that checks if command has been edited manually, if so it will turn autoShell off
+      atom.config.observe("terminal.shell", () => {
+        localStorage.setItem("terminal.autoShell", "false")
+      }),
```

Comment:
The remaining issue is that when the user edits the shell config and then unsets it. autoShell becomes false forever. This means if they unset the shell command, we can't detect it.

Reply:
Using the folllowing does not help. newValue does not become `""` even if the user unsets the value 
```ts
      atom.config.onDidChange("terminal.shell", ({ newValue }) => {
        if (newValue == "") {
          localStorage.setItem("terminal.autoShell", "true")
        } else {
          localStorage.setItem("terminal.autoShell", "false")
        }
```

Reply:
To fix this, I think we need to either: 
- extend configuration manager of Atom:
 	- allow setting default values asynchronously
	- add support for `onDidUnset`

or
- make a config and call it `resetCache`, which allows users to reset the cache.

Reply:
I think to make things easy we can just set `terminal.shell` when first installed then assume that if the user changes it they know what they are doing.

Reply:
Yes, we can assume that for now.

Reply:
This logic is a lot more complex than we need to just set the shell at install.